graphql/src/test/java/com/linecorp/armeria/server/graphql/ExecutionIdGeneratorTest.java

```diff
@@ -0,0 +1,143 @@
+            executionId = executionContext.getExecutionId();
+            return super.execute(executionContext, parameters);
+        }
+    }
+
+    static class ConcatExecutionIdGenerator implements ExecutionIdGenerator {
+        @Override
+        public ExecutionId generate(ServiceRequestContext requestContext, String query, String operationName,
+                                    GraphQLContext graphqlContext) {
+            return ExecutionId.from(requestContext + query + operationName + graphqlContext);
+        }
+    }
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/test.graphqls").toURI());
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            final GraphqlService customizedExecutionIdService =
+                    GraphqlService.builder()
+                                  .executionIdGenerator(new ConcatExecutionIdGenerator())
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            sb.service("/graphql", service)
+              .service("/graphql-customized-execution-id", customizedExecutionIdService);
+        }
+    };
+
+    static DataFetcher<String> dataFetcher(String value) {
+        return environment -> {
+            capturedGraphQLContext = environment.getGraphQlContext();
+            final ServiceRequestContext ctx = GraphqlServiceContexts.get(environment);
+            assertThat(ctx.eventLoop().inEventLoop()).isTrue();
+            // Make sure that a ServiceRequestContext is available
+            assertThat(ServiceRequestContext.current()).isSameAs(ctx);
+            return value;
+        };
+    }
+
```

Comment:
What do you think of clearing the `executionId` for each invocation so that test failures are clearer (if they occur)

```
    @BeforeEach
    void beforeEach() {
        idStrategy.executionId = null;
    }
```